pkg/cmd/cli/cmd/wrappers.go

```diff
@@ -185,28 +185,34 @@ const (
 
   # Switch to raw terminal mode, sends stdin to 'bash' in ruby-container from pod 'mypod' and sends stdout/stderr from 'bash' back to the client
   %[1]s exec mypod -c ruby-container -i -t -- bash -il`
+)
 
-	completionLong = `This command prints shell code which must be evaluation to provide interactive
-completion of openshift cli commands.`
+func NewCmdCompletion(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	cmdHelpString := "openshift cli"
+	cmdHelpName := fullName
+
+	if strings.HasSuffix(fullName, "admin") || strings.HasSuffix(fullName, "oadm") {
```

Comment:
This logic looks a little confusing... Can't you just use `fullName` directly? I believe it gives you `openshift`, `openshift cli`, `openshift adm`, `oadm` or `oc`, depending on what's being used, and I think it's what we want.